cmd/cloudshell_open/main.go

```diff
@@ -214,24 +214,36 @@ func run(c *cli.Context) error {
 	serviceName = tryFixServiceName(serviceName)
 
 	image := fmt.Sprintf("gcr.io/%s/%s", project, serviceName)
-	fmt.Println(infoPrefix + " FYI, running the following command:")
-	cmdColor.Printf("\tdocker build -t %s %s\n", parameter(image), parameter("."))
 
 	end = logProgress(fmt.Sprintf("Building container image %s", highlight(image)),
 		fmt.Sprintf("Built container image %s", highlight(image)),
 		"Failed to build container image.")
-	err = build(appDir, image)
-	end(err == nil)
+
+	exists, err := dockerFileExists(appDir)
 	if err != nil {
 		return err
 	}
 
+	if exists {
+		fmt.Println(infoPrefix + " FYI, running the following command:")
+		cmdColor.Printf("\tdocker build -t %s %s\n", parameter(image), parameter("."))
+		err = dockerBuild(appDir, image)
+	} else {
+		fmt.Println(infoPrefix + " FYI, running the following command:")
+		cmdColor.Printf("\tpack build %s --path %s --builder heroku/buildacks\n", parameter(image), parameter(appDir))
+		err = packBuild(appDir, image)
+	}
+	end(err == nil)
+	if err != nil {
+		return fmt.Errorf("failed to build image: %s", err)
```

Comment:
can we do this:

    this application doesn't have a Dockerfile, attempted to build it as a buildpack and failed: %v

Reply:
PTAL